generators/java/generator-utils/src/main/java/com/fern/java/AbstractPoetClassNameFactory.java

```diff
@@ -34,13 +37,25 @@ public final ClassName getRootClassName(String className) {
     }
 
     public final String getCorePackage() {
-        List<String> tokens = new ArrayList<>(packagePrefixTokens);
-        tokens.add("core");
-        return String.join(".", tokens);
+        switch (packageLayout) {
+            case FLAT:
+                return String.join(".", packagePrefixTokens);
+            case NESTED:
+            default:
+                List<String> tokens = new ArrayList<>(packagePrefixTokens);
+                tokens.add("core");
+                return String.join(".", tokens);
+        }
     }
 
     public final String getPaginationPackage() {
-        return getCorePackage() + ".pagination";
+        switch (packageLayout) {
+            case FLAT:
+                return getCorePackage();
```

Comment:
We _might_ want to keep the nested packages within `core`, but if this is compatible as-is, then this might be a nice improvement too.